indexer/editable_map_object.cpp

```diff
@@ -744,6 +766,125 @@ bool EditableMapObject::ValidateWebsite(string const & site)
   return true;
 }
 
+//static
+bool EditableMapObject::ValidateFacebookPage(string const & page)
+{
+  if (page.empty())
+    return true;
+  // Check that facebookPage contains valid username.
+  // See rules: https://www.facebook.com/help/105399436216001
+  if (regex_match(page, regex(R"(^@?[a-zA-Z\d.\-]{5,}$)")))
+    return true;
+  if (ValidateWebsite(page))
+  {
+    string facebookPageUrl = page;
+    // Check if HTTP protocol is present
+    if (!strings::StartsWith(page, "http://") && !strings::StartsWith(page, "https://"))
+      facebookPageUrl = "https://" + page;
+
+    const url::Url url = url::Url(facebookPageUrl);
+    const string &domain = strings::MakeLowerCase(url.GetWebDomain());
+    // Check Facebook domain name
+    return strings::EndsWith(domain, "facebook.com") || strings::EndsWith(domain, "fb.com")
+        || strings::EndsWith(domain, "fb.me") || strings::EndsWith(domain, "facebook.de")
+        || strings::EndsWith(domain, "facebook.fr");
+  }
+
+  return false;
+}
+
+//static
+bool EditableMapObject::ValidateInstagramPage(string const & page)
+{
+  if (page.empty())
+    return true;
+
+  // Check that `page` contains valid username.
+  // Rules took here: https://blog.jstassen.com/2016/03/code-regex-for-instagram-username-and-hashtags/
+  if (regex_match(page, regex(R"(^@?[A-Za-z0-9_][A-Za-z0-9_.]{0,28}[A-Za-z0-9_]$)")))
```

Comment:
Please cache all regexes in local or global (but in an anonymous namespace) `static regex const re = …;` variables for speed.